bson-kotlin/src/main/kotlin/org/bson/codecs/kotlin/DataClassCodecProvider.kt

```diff
@@ -18,9 +18,13 @@ package org.bson.codecs.kotlin
 import org.bson.codecs.Codec
 import org.bson.codecs.configuration.CodecProvider
 import org.bson.codecs.configuration.CodecRegistry
+import java.lang.reflect.Type
 
 /** A Kotlin reflection based Codec Provider for data classes */
 public class DataClassCodecProvider : CodecProvider {
     override fun <T : Any> get(clazz: Class<T>, registry: CodecRegistry): Codec<T>? =
-        DataClassCodec.create(clazz.kotlin, registry)
+        get(clazz, emptyList(), registry)
+
+    override fun <T : Any> get(clazz: Class<T>, typeArguments: List<Type>, registry: CodecRegistry): Codec<T>? =
+        DataClassCodec.create(clazz.kotlin, registry, typeArguments)
```

Comment:
If this provider is rolled back to the previous implementation, some tests in `DataClassCodecTest` (for example, `testDataClassWithParameterizedDataClass`), which this PR did not modify, start failing with `CodecConfigurationException: Could not find codec for number with type N`. I.e., they do test parameterization and it appear to work when the provider properly overrides the new `get` method.